File: VB/DxSample/Form1.vb
Imports DevExpress.XtraGrid.Views.Base
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace DxSample

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub OnValidateRow(ByVal sender As Object, ByVal e As ValidateRowEventArgs)
            Dim p As Person = CType(e.Row, Person)
            e.Valid = String.IsNullOrEmpty(p.Name) OrElse p.Name.Contains("A")
        End Sub
    End Class
End Namespace

File: CS/DxSample/Form1.cs
using DevExpress.Utils;
using DevExpress.XtraGrid.Views.Base;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DxSample {
    public partial class Form1 :Form {
        public Form1() {
            InitializeComponent();
        }

        private void OnValidateRow(object sender, ValidateRowEventArgs e) {
            Person p = (Person)e.Row;
            e.Valid = string.IsNullOrEmpty(p.Name) || p.Name.Contains("A");
        }
    }
}
